src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -402,13 +402,16 @@ private void ProcessTraceroute(string targetNameOrAddress)
                             }
                         }
 #endif
+
                         reply = SendCancellablePing(hopAddress, timeout, buffer, pingOptions, timer);
 
                         if (!Quiet.IsPresent)
                         {
                             var status = new PingStatus(
                                 Source,
-                                routerName,
+                                routerName != string.Empty
+                                    ? routerName
+                                    : hopAddressString,
```

Comment:
I wonder why do you not use InitProcessPing()?

Reply:
Hm, that is a very good point. It looks like I completely skip it if ResolveDestination isn't set... but InitProcessPing already checks that, so... yeah it makes more sense to just use that method.

Good catch, thank you!

Reply:
Done! Changed the earlier logic to always run through `InitProcessPing()`.

I have it doing a retry if the initial DNS lookup fails when using `-ResolveDestination` since these failures can sometimes be intermittent / latency based.